### PR TITLE
Switch to simplesam for BAM parsing

### PR DESCRIPTION
**Pull request by mr-c**

### CODE REVIEW

**Comment by mr-c** [comment]
- [ ] Is it mergeable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Is the Copyright year up to date?

**Comment by mr-c** [comment]
@ctb ready for review & merge

**Comment by ctb** [comment]
Is this for 2.0?

**Comment by mr-c** [comment]
The Makefile didn't work before, so yes.

File: setup.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def check_for_openmp():
         "url": 'https://khmer.readthedocs.org/',
         "packages": ['khmer', 'khmer.tests', 'oxli'],
         "package_dir": {'khmer.tests': 'tests'},
-        "install_requires": ['screed >= 0.9', 'bz2file'],
+        "install_requires": ['screed >= 0.9', 'bz2file', 'simplesam'],
```
Comment (ctb) [inline setup.py]: I'm -0 on this as a requirement; this is all for a sandbox script that most people won't use. Can we put it in extras or somewhere else?

**Comment by ctb** [comment]
So I'll just tag that 2.0, then.

**Comment by mr-c** [comment]
It could be moved to the reqs for running the tests or moved out of the
sandbox

On Wed, Aug 5, 2015 at 10:09 AM C. Titus Brown <notifications@github.com>
wrote:

> So I'll just tag that 2.0, then.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1228#issuecomment-128073455>.
>
-- 
Michael R. Crusoe: Programmer & Bioinformatician crusoe@ucdavis.edu
The lab for Data Intensive Biology; University of California, Davis
https://impactstory.org/MichaelRCrusoe http://twitter.com/biocrusoe

**Comment by ctb** [comment]
On Wed, Aug 05, 2015 at 10:10:49AM -0700, Michael R. Crusoe wrote:
> It could be moved to the reqs for running the tests or moved out of the
> sandbox

Well, the latter is not going to happen for 2.0, I think.  Up to you.

**Comment by mr-c** [comment]
- [ ] Is it mergeable?
- [ ] `make test` Did it pass the tests?
- [ ] `make clean diff-cover` If it introduces new functionality in
  `scripts/` is it tested?
- [ ] `make format diff_pylint_report cppcheck doc` Is it well formatted?
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the `ChangeLog`?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Do the changes respect streaming IO? (Are they
  tested for streaming IO?)
- [ ] Is the Copyright year up to date?

**Comment by mr-c** [comment]
Simplesam has been moved to an extra; the PR rebased, tests pass: ready for review & merge @ctb @luizirber @camillescott @standage

File: Makefile [hunk 2]
```diff
@@ -194,7 +194,7 @@ pylint: $(PYSOURCES) $(wildcard tests/*.py)
 		setup.py khmer/[!_]*.py khmer/__init__.py scripts/*.py tests \
 		oxli/*.py || true
 
-pylint_report.txt: ${PYSOURCES} $(wildcard tests/*.py)
+pylint_report.txt: ${PYSOURCES} $(wildcard tests/*.py) $(wildcard sandbox/*.py)
```
Comment (ctb) [inline Makefile]: Should this be part of this PR? I don't see any harm in it, just wanted to check.

Comment (mr-c) [inline Makefile]: Yes, otherwise `make diff_pylint_report.txt` won't pick up changes made to `sandbox/*.py` scripts since the last run.

**Comment by ctb** [comment]
LGTM - thanks!

**Comment by ctb** [comment]
(working on merging in master)

**Comment by ctb** [comment]
thx @mr-c!

**Comment by mr-c** [comment]
Thank you, @ctb!

On Sun, May 8, 2016 at 3:33 PM, C. Titus Brown <notifications@github.com>
wrote:

> thx @mr-c <https://github.com/mr-c>!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1228#issuecomment-217718872>
>



-- 
Michael R. Crusoe                          michael.crusoe@gmail.com
Community Engineer                 Common Workflow Language project
https://impactstory.org/u/0000-0002-2961-9670   +32 (0) 2 808 25 58
                                                    +1 480 627 9108

**Pull request merged**